pkg/resources/statefulsets/minio-statefulset.go

```diff
@@ -132,6 +132,14 @@ func minioEnvironmentVars(t *miniov2.Tenant, skipEnvVars map[string][]byte, opVe
 	serverURL := t.MinIOServerEndpoint()
 	if t.HasMinIODomains() {
 		serverURL = t.Spec.Features.Domains.Minio[0]
+		// Provide adequate schema to server URL if a list of domains is available
+		if !strings.HasPrefix(serverURL, "http") {
+			useSchema := "http"
+			if t.TLS() {
+				useSchema = "https"
+			}
+			serverURL = fmt.Sprintf("%s://%s", useSchema, serverURL)
+		}
```

Comment:
Enforce schema via `t.TLS()` there is no reason anyone should provide `http://` as part of the domains input YAML, they have been named appropriately for this reason as "DOMAIN" only not a URL.

The following code that is all is needed.
```go
scheme := "http"
if t.TLS() {
   scheme = "https"
}
serverURL = fmt.Sprintf("%s://%s", scheme, t.Spec.Features.Domains.Minio[0])
```

Reply:
Modified code to suit

Reply:
there's a possibility the tenant might be running without TLS but is exposed via an ingress with TLS, that's why the list of domains took in the schema, also internal + external domains might be a mix of https and http domains, so I think we should still be defensive about this list containing schemas and removing them from the result list of domains

Reply:
Yeah for that relying on domain to pre-populate the scheme logic is not correct.

Because if you see the code also at random picks the first domain to be the incoming domain for server URL.

These kind of automatic internal decisions are complicated to remember.

Thats why we shouldn't overload domain input to handle multiple cases. Domain list is only meant to enable virtual DNS for buckets.

It is better to provide another variable that indicates clearly what the input is and what customers must provide.